### PR TITLE
Templatize via Cookiecutter for enhanced audience utility

### LINKED ISSUE #4: Add common template change functionality for derivative implementations
This issue outlines adding functionality which enables common template changes within derivative implementations (those based on this template).

> - Is there a way to auto-fill key template components? i.e. Some variables are generic and need to be specified for each new template spawn. Maybe if auto-fill is too difficult, a CHANGEME.md file might specify which variables to update with a new spawn. 🤷 
...

_Originally posted by @gwaybio in https://github.com/WayScience/gc-assayworks-bucket/pull/1#pullrequestreview-1271465245_

### PR DESCRIPTION
**Pull request by d33bs**
This PR addresses #4 by moving to a [Cookiecutter](https://github.com/cookiecutter/cookiecutter) implementation for this repository. The benefit of this change entails making this template more accessible through Cookiecutter commands and more efficient through common text replacements which would otherwise need to be performed manually. 

Note: After this is merged I plan to remove the Github template status (which will no longer be as helpful or accurate to the intent).

A demonstration of the audience experience after a merge can be found from the following command:
```shell
python -m cookiecutter https://github.com/d33bs/gc-cloud-storage-bucket.git --checkout templatize-via-cookiecutter
```

Thanks in advance for any thoughts and comments you may have!

Closes #4

### CODE REVIEW

**Review by falquaddoomi — COMMENTED** [review]
I can see the value in using cookiecutter to customize the names of files and strings in documentation. In the case of Terraform config, I'd personally opt for one of Terraform's existing methods for setting input variables (https://developer.hashicorp.com/terraform/language/values/variables#assigning-values-to-root-module-variables) rather than writing them into the default value fields within your modules. Typically, you'd create a `terraform.tfvars` file in the folder where you expect your users to invoke Terraform and instruct them to modify it; you could also populate it with cookiecutter.

**Comment by d33bs** [comment]
Thank you @falquaddoomi ! I've added a `terraform.tfvars` file as you suggested and along with other updates.

**Review by vincerubinetti — APPROVED** [review]
Reviewed writing, LGTM

**Review by falquaddoomi — COMMENTED** [review]
Oof, sorry, forgot to submit this review. There's just the one thing about the `.tfvars` file needing to be renamed to something TF will automatically use, but otherwise this all looks great!

File: {{ cookiecutter.project_name }}/variables.tfvars
[the remark this answers is not in the document]
Comment (d33bs) [reply {{ cookiecutter.project_name }}/variables.tfvars:1]: Thanks for this, apologies about the mistake! I am making some updates here to add a test for this case and making the relevant additions in further commits.

[the remark this answers is not in the document]
Comment (d33bs) [reply {{ cookiecutter.project_name }}/variables.tfvars:1]: Added this in and the test I mentioned.

**Review by falquaddoomi — APPROVED** [review]
Looks good! Thanks for taking my previous suggestions into consideration.

**Comment by d33bs** [comment]
Thank you @falquaddoomi and @vincerubinetti for your reviews! Merging this in now.

**Pull request merged**